components/scream/src/physics/spa/atmosphere_prescribed_aerosol.cpp

```diff
@@ -0,0 +1,216 @@
+#include "atmosphere_prescribed_aerosol.hpp"
+
+#include "ekat/ekat_assert.hpp"
+#include "ekat/util/ekat_units.hpp"
+
+#include <array>
+
+namespace scream
+{
+  using namespace spa;
+// =========================================================================================
+SPA::SPA (const ekat::Comm& comm, const ekat::ParameterList& params)
+ : m_spa_comm (comm)
+ , m_spa_params (params)
+{
+  // Nothing to do here
+}
+
+// =========================================================================================
+void SPA::set_grids(const std::shared_ptr<const GridsManager> grids_manager)
+{
+  using namespace ekat::units;
+  using namespace ShortFieldTagsNames;
+
+  // The units of mixing ratio Q are technically non-dimensional.
+  // Nevertheless, for output reasons, we like to see 'kg/kg'.
+  auto Q = kg/kg;
+  Q.set_string("kg/kg");
+  Units nondim(0,0,0,0,0,0,0);
+
+  const auto& grid_name = m_spa_params.get<std::string>("Grid");
+  auto grid  = grids_manager->get_grid(grid_name);
+  m_num_cols = grid->get_num_local_dofs(); // Number of columns on this rank
+  m_num_levs = grid->get_num_vertical_levels();  // Number of levels per column
+
+  // Define the different field layouts that will be used for this process
+
+  // Layout for 3D (2d horiz X 1d vertical) variable defined at mid-level and interfaces 
+  FieldLayout scalar3d_layout_mid { {COL,LEV}, {m_num_cols, m_num_levs} };
+  FieldLayout scalar2d_layout     { {COL},     {m_num_cols} };
+  FieldLayout scalar1d_layout_mid { {LEV},     {m_num_levs} };
+  // Use VAR field tag for gases for now; consider adding a tag?
+  FieldLayout scalar3d_swband_layout { {COL,SWBND, LEV}, {m_num_cols, m_nswbands, m_num_levs} }; 
+  FieldLayout scalar3d_lwband_layout { {COL,LWBND, LEV}, {m_num_cols, m_nlwbands, m_num_levs} }; 
+
+  // Set of fields used strictly as input
+  constexpr int ps = Pack::n;
+  add_field<Required>("p_mid"      , scalar3d_layout_mid, 1/kg,   grid_name, ps);
+  add_field<Required>("hyam"       , scalar1d_layout_mid, nondim, grid_name, ps); // TODO: These fields should  be loaded from file and not registered with the field manager.
+  add_field<Required>("hybm"       , scalar1d_layout_mid, nondim, grid_name, ps); // TODO: These fields should  be loaded from file and not registered with the field manager.
+
+  add_field<Required>("PS_beg"     , scalar2d_layout,     Pa,     grid_name, ps); // TODO: These fields should  be loaded from file and not registered with the field manager.
```

Comment:
We could wait, the idea here was to get a version of SPA that works in case we have parallel tasks that could benefit from SPA being at least able to be run in some way, while we wait on the SCORPIO changes we need to do this correctly.